sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -121,9 +125,19 @@ def work
               end
 
               export_batch(batch)
+              reset_on_fork(restart_thread: false)
```

Comment:
This should move up inside the `batch = lock do ... end` block.

Reply:
Have made the corrections.